Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift

```diff
@@ -543,7 +543,7 @@ final class SchemeDescriptorsGenerator: SchemeDescriptorsGenerating {
         guard let pbxTarget = generatedProject.targets[targetReference.name] else { return nil }
         let relativeXcodeProjectPath = resolveRelativeProjectPath(targetNode: target,
                                                                   generatedProject: generatedProject,
-                                                                  rootPath: rootPath)
+                                                                  rootPath: target.project.xcodeProjPath)
```

Comment:
`rootPath` is correct here, we need to change an upstream caller to propagate the correct root path - which needs to be `xcodeProjPath.parentDirectory` to fix the issue you were facing.

```diff
         try project.schemes.map { scheme in
             try generateScheme(scheme: scheme,
-                               path: project.path,
+                               path: project.xcodeProjPath.parentDirectory,
                                graph: graph,
                                generatedProjects: [project.path: generatedProject])
         }
```